### PR TITLE
feat: leaderboard page integration with backend APIs

### LINKED ISSUE #21: create leaderboard page ui
- This page will be visible for Guest as well as signed in user
- Signed in user will also be shown his personal high score
- Guest user will be shown only the global leading scores

### PR DESCRIPTION
**Pull request by ravikp7**
fixes #21

### CODE REVIEW

**Comment by ravikp7** [comment]
Visual changes shown in this loom video:
https://www.useloom.com/share/46cb0f189e7541a597e0d969eee196dd

File: client/src/App.js [hunk 1]
```diff
@@ -16,7 +19,11 @@ const App = () => (
         <Route exact path="/signin" component={SignIn} />
         <Route exact path="/signup" component={SignUp} />
         <Route exact path="/forgotPassword" component={ForgotPassword} />
-        <Route exact path="/leaderboard" component={LeaderboardPage} />
+        <Route
+          exact
+          path="/leaderboard"
+          render={routeProps => <LeaderboardPage {...{ userId }} {...routeProps} />}
```
Comment (nimish-gupta) [inline client/src/App.js:25]: Is leaderboard access to everyone or only to authenticate users?

**Review by nimish-gupta — COMMENTED** [review]
Lgtm

File: client/src/App.js (hunk 1 above)
Comment (ravikp7) [inline client/src/App.js:25]: - It's here to let the team know how to use the component.
- It'll be removed later

Comment (ravikp7) [reply client/src/App.js:25]: - It's here to let the team know how to use the component later.
- It'll be removed later

**Pull request merged**